third_party/amd/backend/compiler.py

```diff
@@ -133,36 +112,35 @@ def make_ttir(mod, metadata, opt):
         return mod
 
     @staticmethod
-    def make_ttgir(mod, metadata, opt):
+    def make_ttgir(mod, metadata, options):
         pm = ir.pass_manager(mod.context)
         pm.enable_debug()
-        # TODO: capability
-        passes.ttir.add_convert_to_ttgpuir(pm, opt.num_warps, opt.warp_size, opt.num_ctas, 90)
+        passes.ttir.add_convert_to_ttgpuir(pm, options.num_warps, options.warp_size, options.num_ctas, None)
         pm.run(mod)
         pm = ir.pass_manager(mod.context)
         pm.enable_debug()
         passes.ttgpuir.add_coalesce(pm)
         amd.passes.ttgpuir.add_remove_layout_conversions(pm)
         passes.ttgpuir.add_optimize_thread_locality(pm)
-        amd.passes.ttgpuir.add_accelerate_matmul(pm, opt.arch, opt.matrix_instr_nonkdim, opt.kpack)
+        amd.passes.ttgpuir.add_accelerate_matmul(pm, options.arch, options.matrix_instr_nonkdim, options.kpack)
         amd.passes.ttgpuir.add_remove_layout_conversions(pm)
         amd.passes.ttgpuir.add_optimize_epilogue(pm)
         passes.ttgpuir.add_optimize_dot_operands(pm)
-        if opt.num_stages == 0 and opt.has_amd_mma_instr():
+        if options.num_stages == 0 and amd.has_matrix_core_feature(options.arch):
```

Comment:
This is where this `has_matrix_core_feature` is used. IIRC, this is a legacy code from our fork. At that time, we needed to check two things: is_hip() and num_stages=0, to enable this stream-pipeline pass. 
Now I think we only need to check num_stages=0. The stream-pipeline pass should work (in theory) for archs without mfma or wmma support.

Reply:
Ha okay. Yeah we should try it out. We can keep this as and have a follow up patch to drop if so. If we can remove that'd be great!